### DISCUSSION TITLE
Add Zoom and Pan booleans to canvas class

### DISCUSSION THREAD

**Discussion by enriqueerrando** [fabricjs/fabric.js#9002] [Ideas]
Hi,

Since zoomming and panning are pretty common functionality that work pretty much the same way in every case, instead of us having to code them almost from scratch, it would be great if there was a simple boolean to enable/disable them, which would simplify everything soooooo much and, ofcourse, unclutter our code.

Thanks in advance!

**Comment by ShaMan123** [comment]
boolean definitely not - we must support subclassing, customization and code scoping
but a handler/interaction manager/event - I am for

**Reply by enriqueerrando** [reply]
Anything that simplifies zoomming and panning works for me, hahahaha!!
